### DISCUSSION TITLE
Q-toggle does not work inside composition API

### DISCUSSION THREAD

**Discussion by NikolaNbgd** [quasarframework/quasar#12328] [General - Components / Directives / etc] [answered]
Hi, does anyone know why <q-toggle> does not work inside composition API?
Here is the code
```
<q-toggle
    :model-value="data"
    right-label
 />
```

and down bellow inside the setup function I defined const data = false;

I don't see a mistake here, but still, it does not change the value on click. Thanks

EDIT: I found what is the problem, I literally need to use v-model instead of  :model-value

**Comment by MilosPaunovic** [comment]
You can check more examples on the docs page:
[https://quasar.dev/vue-components/toggle](https://quasar.dev/vue-components/toggle)

And to check out Codepens, to see it in action:
[https://codepen.io/milospaunovic/pen/XWzKEgQ?editors=101](https://codepen.io/milospaunovic/pen/XWzKEgQ?editors=101)

**Comment by rstoenescu** [comment, marked as answer]
`v-model` is alias to `:model-value="model" @update:model-value="val => { model = val }"`.
What you tried is just supplying the value for the model, but no way to update it when QToggle emits the change.
